File: Searching_Algorithms/Python/linearSearch.py
def linear_search(arr, target):
    # Iterate through the array element by element.
    for i in range(len(arr)):
        # If the current element is equal to the target, return its index.
        if arr[i] == target:
            return i
    
    # If the target is not found in the array, return -1 to indicate it's not present.
    return -1

# Example usage:
arr = [68, 33, 25, 15, 22, 11, 80]
target = 25
result = linear_search(arr, target)

if result != -1:
    print(f"Element {target} found at index {result}")
else:
    print(f"Element {target} not found in the array")

File: Searching_Algorithms/C/linearSearch.c
#include <stdio.h>

int linearSearch(int array[], int n, int x)
{

    // Going through array sequencially
    for (int i = 0; i < n; i++)
        if (array[i] == x)
            return i;
    return -1;
}

int main()
{
    int array[10], size;
    int key, i;
    printf("Enter the size of the array:");
    scanf("%d",&size);
    printf("Linear Search:\n");
    printf("Enter Array Elements:");
    for (i = 0; i < size; i++)
    {
        scanf("%d", &array[i]);
    }
    printf("Enter the element to be searched:");
    scanf("%d", &key);
    int result = linearSearch(array, size, key);

    (result == -1) ? printf("Element not found\n") : printf("Element found at index: %d\n", result);
}